bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/DanfossAirUnit.java

```diff
@@ -0,0 +1,278 @@
+ * {@link DanfossAirUnitCommunicationController}
+ *
+ * @author Ralf Duckstein - Initial contribution
+ * @author Robert Bach - heavy refactorings
+ */
+
+@SuppressWarnings("SameParameterValue")
+@NonNullByDefault
+public class DanfossAirUnit implements Commands {
+
+    private final Logger logger = LoggerFactory.getLogger(DanfossAirUnit.class);
+    private final DanfossAirUnitCommunicationController communicationController;
+
+    public DanfossAirUnit(InetAddress inetAddr, int port) {
+        this.communicationController  = new DanfossAirUnitCommunicationController(inetAddr, port);
+    }
+
+    public void cleanUp() {
+        this.communicationController.disconnect();
+    }
+
+    private boolean getBoolean(byte[] operation, byte[] register) throws IOException {
+        return communicationController.sendRobustRequest(operation, register)[0] != 0;
+    }
+
+    private void setSetting(byte[] register, boolean value) throws IOException {
+        setSetting(register, value ? (byte) 1 : (byte) 0);
+    }
+
+    private short getWord(byte[] operation, byte[] register) throws IOException {
+        byte[] resultBytes = communicationController.sendRobustRequest(operation, register);
+        return (short)((resultBytes[0] << 8)  | (resultBytes[1] & 0xFF));
+    }
+
+    private byte getByte(byte[] operation, byte[] register) throws IOException {
+        return communicationController.sendRobustRequest(operation, register)[0];
+    }
+
+    private String getString(byte[] operation, byte[] register) throws IOException {
+        // length of the string is stored in the first byte
+        byte[] result = communicationController.sendRobustRequest(operation, register);
+        return new String(result, 1, result[0], StandardCharsets.US_ASCII);
+    }
+
+    private void set(byte[] operation, byte[] register, byte value) throws IOException {
+        byte[] valueArray = { value };
+        communicationController.sendRobustRequest(operation, register, valueArray);
+    }
+
+    private void set(byte[] operation, byte[] register, short value) throws IOException {
+        communicationController.sendRobustRequest(operation, register, shortToBytes(value));
+    }
+
+    private byte[] shortToBytes(short s) {
+        return new byte[]{(byte)((s & 0xFF00) >> 8),(byte)(s & 0x00FF)};
+    }
+
+    private short getShort(byte[] operation, byte[] register) throws IOException {
+        byte[] result = communicationController.sendRobustRequest(operation, register);
+        return (short) ((result[0] << 8) + (result[1] & 0xff));
+    }
+
+    private float getTemperature(byte[] operation, byte[] register) throws IOException {
+        short shortTemp = getShort(operation, register);
+        float temp = ((float) shortTemp) / 100;
+        if (temp <= -274 || temp > 100) {
+            throw new IllegalArgumentException(String.format("Invalid temperature: %s", temp));
```

Comment:
IllegalArgumentException should be reserved for invalid or incorrect arguments to a function or constructor. This case is neither of those as you are handling a response message. You should either throw an IOException or return `Float.NaN` then handle the NaN case elsewhere in your code.